components/usage-api/usage/v1/usage.proto

```diff
@@ -15,6 +15,66 @@ service UsageService {
 
     // GetCostCenter retrieves the spending limit with its associated attributionID
     rpc GetCostCenter(GetCostCenterRequest) returns (GetCostCenterResponse) {}
+
+    // Triggers reconciliation of usage with ledger implementation.
+    rpc ReconcileUsageWithLedger(ReconcileUsageWithLedgerRequest) returns (ReconcileUsageWithLedgerResponse) {}
+
+    // Creates, or updates, a Usage record
+    rpc UpsertUsage(UpsertUsageRequest) returns (UpsertUsageResponse) {}
+}
+
+enum UsageKind {
+    USAGE_KIND_UNKNOWN = 0;
+    USAGE_KIND_WORKSPACE_INSTANCE = 1;
+    USAGE_KIND_INVOICE = 2;
+}
+
+message Usage {
+    // UUID identifier of the Usage event. Required.
+    string id = 1;
+
+    // Attribution ID for the usage record. Required.
+    string attribution_id = 2;
+
+    // Description of the Usage event for humans.
+    string description = 3;
+
+    // Positive, or negative amount of credits for the Usage event.
+    // Credits are accepted with full precision, but returned with two decimal point precision.
+    double credits = 4;
```

Comment:
Should be `sint64` and `creditCents` ([see](https://github.com/gitpod-io/gitpod/pull/12584#discussion_r960779141)).

Reply:
The intention here is not to mirror the table. In the table we can store it as credits * 100 rounded to nearest int but for usability from the API, I'd keep it as credits here (also see the comment above). This will avoid translating the credits everywhere into credit cents and back.